choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerExecutorDBService.java

```diff
@@ -0,0 +1,279 @@
+		
+		final int validatedPage = page < 0 ? 0 : page;
+		final int validatedSize = size <= 0 ? Integer.MAX_VALUE : size;
+		final Direction validatedDirection = direction == null ? Direction.ASC : direction;
+		final String validatedSortField = Utilities.isEmpty(sortField) ? CoreCommonConstants.COMMON_FIELD_NAME_ID : sortField.trim();
+		
+		if (!ChoreographerExecutor.SORTABLE_FIELDS_BY.contains(validatedSortField)) {
+			throw new InvalidParameterException("Sortable field with reference '" + validatedSortField + "' is not available");
+		}
+		
+		try {
+			return executorRepository.findAll(PageRequest.of(validatedPage, validatedSize, validatedDirection, validatedSortField));
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public Optional<ChoreographerExecutorResponseDTO> getExecutorOptionalByIdResponse(final long id) { //TODO junit
+		logger.debug("getExecutorOptionalByIdResponse started...");
+		
+		final Optional<ChoreographerExecutor> optional = getExecutorOptionalById(id);
+		if (optional.isEmpty()) {
+			return Optional.empty();
+		}
+		
+		final ChoreographerExecutorResponseDTO dto = DTOConverter.convertExecutorToExecutorResponseDTO(optional.get(), executorServiceDefinitionRepository.findAllByExecutor(optional.get()));
+		return Optional.of(dto);
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public Optional<ChoreographerExecutor> getExecutorOptionalById(final long id) { //TODO junit
+		logger.debug("getExecutorById started...");
+		
+		try {
+			return executorRepository.findById(id);	
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public Optional<ChoreographerExecutor> getExecutorOptionalByAddressAndPortAndBaseUri(final String address, final int port, final String baseUri) { //TODO junit
+		logger.debug("getExecutorById started...");
+		Assert.isTrue(!Utilities.isEmpty(address), "address is empty");
+		
+		try {
+			return executorRepository.findByAddressAndPortAndBaseUri(address.toLowerCase().trim(), port, Utilities.isEmpty(baseUri) ? "" : baseUri.trim());	
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+    public void deleteExecutorById(final long id) { //TODO junit
+		logger.debug("deleteExecutorById started...");
+		
+		try {
```

Comment:
you shouldn't allow to delete an executor when there is a session which uses it.